ios/VisionCameraPluginBarcodeScanner.swift

```diff
@@ -47,16 +48,66 @@ class VisionCameraPluginBarcodeScanner: NSObject, FrameProcessorPluginBase {
     return barcodeData
   }
     
-  @objc
-  public static func callback(_ frame: Frame!, withArgs args: [Any]!) -> Any! {
-    let image = VisionImage(buffer: frame.buffer)
-    image.orientation = frame.orientation
-    let format = args[0] as? Int
-    let barcodeOptions = getScannerOptions(scannerFormat: format)
-    guard let barcodeResults = getScannerResults(image: image, barcodeScanner: BarcodeScanner.barcodeScanner(options: barcodeOptions)) else {
+    private static func getImageSize(sampleBuffer: CMSampleBuffer) -> CGSize {
+        let imageBuffer = CMSampleBufferGetImageBuffer(sampleBuffer);
+        let imageWidth  = CVPixelBufferGetWidth(imageBuffer!);
+        let imageHeight = CVPixelBufferGetHeight(imageBuffer!);
+        return CGSizeMake(CGFloat(imageWidth), CGFloat(imageHeight))
+    }
+
+    private static func scaleImage(sampleBuffer: CMSampleBuffer) -> CMSampleBuffer? {
+        let imageBuffer = CMSampleBufferGetImageBuffer(sampleBuffer)!
+        let image = CIImage(cvPixelBuffer: imageBuffer)
+        
+        let scaleFilter = CIFilter(name: "CILanczosScaleTransform")
+        scaleFilter?.setValue(image, forKey: kCIInputImageKey)
+        scaleFilter?.setValue([1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.7].randomElement(), forKey: kCIInputScaleKey)
```

Comment:
What is the background for picking a random scale factor? Does it work through all from 1.1 up to 1.7 in order to try a few different zoom levels?

Reply:
> What is the background for picking a random scale factor? Does it work through all from 1.1 up to 1.7 in order to try a few different zoom levels?

Yeah it tries different zoom levels which I found more effective since we cant really determine how far the barcode is away from the screen. I noticed randomizing how much we scale the barcode up resulted in better results.

Reply:
I see, so each frame gets a chance of a different zoom factor. Does all of this code run each time or only if the original scan resulted in no data? Just worried with all the docs I read regarding the speed the processor plugins have to do be and unsure how we match up - https://mrousavy.com/react-native-vision-camera/docs/guides/frame-processors-plugins-overview#async-frame-processors-with-event-emitters

Reply:
> I see, so each frame gets a chance of a different zoom factor. Does all of this code run each time or only if the original scan resulted in no data? Just worried with all the docs I read regarding the speed the processor plugins have to do be and unsure how we match up - https://mrousavy.com/react-native-vision-camera/docs/guides/frame-processors-plugins-overview#async-frame-processors-with-event-emitters

It runs each time for each frame. Before it passes it to MLKIT to be analyzed it adds the scale filter to the image.
This runs pretty fast though and isn't an issue because from my testing no frames were being dropped. And speed wasn't an issue at all.